traitsui/extras/demo.py

```diff
@@ -121,7 +120,7 @@ def parse_source(file_name):
         The source code, sans docstring.
     """
     try:
-        with src_open(file_name, 'r') as fh:
+        with io.open(file_name, 'r', encoding='utf-8') as fh:
             source_code = fh.read()
         return extract_docstring_from_source(source_code)
     except Exception:
```

Comment:
Suggestion if you are feeling adventurous: return the traceback as the first item in the tuple below, so in the event of a load failure the "docstring" is the error so we no longer get silent failures on parsing.  It would be better to put it into the "log" tab, but this is a simple improvement which should help avoid this sort of problem in the future.

Reply:
@corranwebster That we really fun to do, I learned a few things, and now parsing doesn't fail silently. Thanks for the suggestion!